### ISSUE TITLE
FAQ: What parameter to set to change the directory location of processing files

### ISSUE THREAD

**Issue by kannanuv** [abria/TeraStitcher#10]
This question probably belongs to FAQ. When I stitching tiles the following files are stored in the source directory. 
xml_import.xml and mdata.bin

What should be the xml parameter specified to change the location of these files?

[tera_stitch_xmls.tar.gz](https://github.com/abria/TeraStitcher/files/520436/tera_stitch_xmls.tar.gz)

**Comment by abria** [comment]
I am not sure I understood your question.

You can safely move the xml files anywhere, and you will still be able to use them as they internally point to the source directory. The `mdata.bin`, however, cannot be moved. Specifically, if you move or delete the `mdata.bin`, it will be automatically regenerated.

**Comment by kannanuv** [comment]
In our data processing, computational pipeline the image data is generated by one user and stitched by another user. Ideally, we want to segregate the data from each part of the pipeline. We want the stitching related files to be stored in the directory of stitched images or some other working directory other than the data source directory. Is there any way to do this by passing appropriate parameters?

**Comment by abria** [comment]
Yes, you can manually provide the path of the generated XMLs using the [`--projout`](https://github.com/abria/TeraStitcher/wiki/User-Interface#--projoutstring) parameter.

**Comment by kannanuv** [comment]
And the null.xml too which is a modified version of TeraStitcher_align.xml previously attached

**Comment by kannanuv** [comment]
ok got it. I shall modify the parameter and test it and close the issue.

**Comment by abria** [comment]
Please remember to use the same xml in the subsequent step. For instance, if you save the xml with `--projout` at step `i`, you should give it in input it with `--projin` to step `i+1`.

**Comment by kannanuv** [comment]
running: /home/kannanuv/Documents/softwares/teraStitcher/TeraStitcher-standalone-1.9.62-Linux/bin/terastitcher --import  --projin="/data/iDISCO_Processing/1675/TeraStitcher_import.xml" --projout="/data/iDISCO_Processing/1675/" 
ERROR: in StackedVolume::saveBinaryMetadata(...): unable to save binary metadata file

It still looks to save the metadata file in the source folder.

**Comment by kannanuv** [comment]
It was the same with specifying xml as the projout

running: /home/kannanuv/Documents/softwares/teraStitcher/TeraStitcher-standalone-1.9.62-Linux/bin/terastitcher --import  --projin="/data/iDISCO_Processing/1675/TeraStitcher_import.xml" --projout="/data/iDISCO_Processing/1675/xml_import.xml" 
ERROR: in StackedVolume::saveBinaryMetadata(...): unable to save binary metadata file

**Comment by kannanuv** [comment]
It looks like the mdata_filepath is generated from the stacks dir and volumePath in vmStackedVolume.cpp

sprintf(mdata_filepath, "%s/%s", stacks_dir, vm::BINARY_METADATA_FILENAME.c_str());
sprintf(mdata_filepath, "%s/%s", volumePath, vm::BINARY_METADATA_FILENAME.c_str());

**Comment by abria** [comment]
As I said, the `mdata.bin` cannot be moved / placed to another directory. If you move or delete it, it will be automatically regenerated. The "_unable to save binary metadata file_" error should indicate that you don't have writing permissions on the source directory.
